<!DOCTYPE html>
<html lang="en-us">
    <head>
        

        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>Transcript Number Rethinking</title>
        
        <style>

    html body {
        font-family: 'Raleway', sans-serif;
        background-color: white;
    }

    :root {
        --accent: red;
        --border-width:  5px ;
    }

</style>


<link rel="stylesheet" href="https://xiangzhangsc.github.io/css/main.css">





<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Raleway">


 <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css"> 


<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css" crossorigin="anonymous">


<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.1/css/all.min.css" integrity="sha512-+4zCK9k+qNFUR5X+cKL9EIR+ZOhtIloNl9GIKS57V1MyNsYpYcUrUeQc9vNfzsWfV28IaLL3i96P9sdNyeRssA==" crossorigin="anonymous" />
 

    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
    
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/go.min.js"></script>
    
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/haskell.min.js"></script>
    
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/kotlin.min.js"></script>
    
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/scala.min.js"></script>
    
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/swift.min.js"></script>
    
    <script>hljs.initHighlightingOnLoad();</script>






<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>


<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>


<script>$(document).on('click', function() { $('.collapse').collapse('hide'); })</script>
 <meta name="generator" content="Hugo 0.82.0" />
        

        

        
            <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        

        

    </head>

    <body>
        

        <nav class="navbar navbar-default navbar-fixed-top">
            <div class="container">
                <div class="navbar-header">
                    <a class="navbar-brand visible-xs" href="#">Transcript Number Rethinking</a>
                    <button class="navbar-toggle" data-target=".navbar-collapse" data-toggle="collapse">
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                </div>
                <div class="collapse navbar-collapse">
                    
                        <ul class="nav navbar-nav">
                            
                                <li><a href="/">Home</a></li>
                            
                                <li><a href="/about/">About</a></li>
                            
                                <li><a href="/post/">Posts</a></li>
                            
                                <li><a href="/project/">Projects</a></li>
                            
                        </ul>
                    
                    
                        <ul class="nav navbar-nav navbar-right">
                            
                                <li class="navbar-icon"><a href="sc.xiang.zhang@gmail.com"><i class="fas fa-envelope"></i></a></li>
                            
                                <li class="navbar-icon"><a href="https://github.com/XiangZhangSC/"><i class="fab fa-github"></i></a></li>
                            
                                <li class="navbar-icon"><a href="https://twitter.com/XiangZhang18/"><i class="fab fa-twitter"></i></a></li>
                            
                                <li class="navbar-icon"><a href="https://www.linkedin.com/in/xiang-zhang-266054185/"><i class="fab fa-linkedin"></i></a></li>
                            
                                <li class="navbar-icon"><a href="https://scholar.google.com/citations?user=p2F-N7gAAAAJ&amp;hl=en"><i class="fab fa-google"></i></a></li>
                            
                        </ul>
                    
                </div>
            </div>
        </nav>


<main>

    <div>
        <h2>Transcript Number Rethinking</h2>
        <h5>May 25, 2021</h5>
        
<a href="https://xiangzhangsc.github.io/tags/count"><kbd class="item-tag">Count</kbd></a>


    </div>

    <div align="start" class="content">
<script src="https://xiangzhangsc.github.io/post/2021-05-25-umi-count-and-transcript-number/index_files/header-attrs/header-attrs.js"></script>


<p>I notice some colleagues working with “transcript count” RNAseq data.
Interestingly, these “transcript count” are not integers.
However, these “transcript count” data were treated as regular count data and still used as input for differential expression analysis tool such as deseq2.</p>
<p>More interestingly, when I asked “what is transcript count?” the answer I got is always “it is UMI count with some sophisticated statistical transformation.”
Although this answer does not contain details, it is clear that “transcript count” is based on UMI count.
What is the “sophisticated statistical transformation?”
With the help of my colleague, I found the details embedded in the publication <span class="citation">(<a href="#ref-Grun2014" role="doc-biblioref">Grün, Kester, and Oudenaarden 2014</a>)</span>.
Surprisingly, the “sophisticated statistical transformation” includes three equations:</p>
<p><span class="math display">\[\begin{align}

K &amp;= k_{o,i} + k_{n,i} \\
(1 - \frac{1}{K})^{m_i} &amp;= \frac{k_{n,i}}{K} \\
m_i &amp;= \frac{ln(\frac{k_{n,i}}{K})}{ln(1 - \frac{1}{K})} \\
&amp;= \frac{ln(\frac{K - k_{o,i}}{K})}{ln(1 - \frac{1}{K})} \\
&amp;= \frac{ln(1 - \frac{k_{o,i}}{K})}{ln(1 - \frac{1}{K})}

\end{align}\]</span></p>
<ol style="list-style-type: decimal">
<li><span class="math inline">\(K\)</span> represents the total number of UMIs.</li>
<li><span class="math inline">\(k_{o,i}\)</span> the number of observed UMIs for gene <span class="math inline">\(i\)</span>.</li>
<li><span class="math inline">\(k_{n,i}\)</span> the number of unobserved UMIs for gene <span class="math inline">\(i\)</span>.</li>
<li><span class="math inline">\(m_i\)</span> the number of transcript molecules derived from gene <span class="math inline">\(i\)</span>.</li>
</ol>
<p>If the sequencing data were generated based on the CEL-Seq protocol <span class="citation">(<a href="#ref-Hashimshony2012" role="doc-biblioref">Hashimshony et al. 2012</a>)</span>, <span class="math inline">\(K = 4^4 = 256\)</span>.
If the sequencing data were generated based on the CEL-Seq2 protocol <span class="citation">(<a href="#ref-Hashimshony2016" role="doc-biblioref">Hashimshony et al. 2016</a>)</span>, <span class="math inline">\(K = 4^6 = 4096\)</span>.
Why <span class="math inline">\(4^4\)</span> or <span class="math inline">\(4^6\)</span>?
This is because UMI is basically a 4 (CEL-Seq) or 6 (CEL-Seq2) base pair nucleotides.
Each position has four possibilities, A, C, T or G.</p>
<p>An important thing to realize is that for every gene, the CEL-seq or CEL-Seq2 only sequences 50 bases of the cDNA.
So, this sequencing method does not distinguish transcript isoformers.
Therefore, here we are talking about gene-level quantification.
For every gene there are <span class="math inline">\(K\)</span> UMIs available for the RNA molecules.
The logic is that the 50 bases tell us from which gene a RNA molecule comes.
The UMI tells us whether this RNA molecule is a PCR artifact or not.</p>
<div id="rethinking-the-conversion-of-umi-count-into-transcript-number" class="section level2">
<h2>Rethinking the conversion of UMI count into transcript number</h2>
<p>The goal of the conversion is to calculate how many RNA molecules are there in the cell at the first place.
To this end, the approach is by asking what is the probability of not observing <span class="math inline">\(k_{n,i}\)</span> UMIs for a gene with <span class="math inline">\(m_i\)</span> copies?</p>
<p>In the <span class="citation">(<a href="#ref-Grun2014" role="doc-biblioref">Grün, Kester, and Oudenaarden 2014</a>)</span>, this probability was given as</p>
<p><span class="math display">\[
(1 - \frac{1}{K})^{m_i} = \frac{k_{n,i}}{K}
\]</span></p>
<p>On the left hand side of the equation it is <span class="math inline">\(m_i\)</span> times Bernoulli probability regarding RNA molecules.
On the right hand side of the equation, it is the proportion of “unobserved” UMI marker for gene <span class="math inline">\(i\)</span>.
My interpretation of this equation is that the author attempted to explain the proportion of “unobserved” UMI marker by <span class="math inline">\(m_i\)</span> times independent Bernoulli process.</p>
<p><strong>I AM STRUGGLING WITH THIS EQUATION!</strong></p>
<p>First of all, why is there an unobserved UMI marker?
Some possibilities:</p>
<ol style="list-style-type: decimal">
<li>a UMI marker is not attached to any RNA molecules.
<ol style="list-style-type: lower-alpha">
<li>No RNA molecule from a gene <span class="math inline">\(i\)</span> (not expressing, <span class="math inline">\(m_i = 0\)</span>).</li>
<li>binding of UMI marker and a RNA molecule fails.</li>
</ol></li>
<li>a UMI marker is attached to a RNA molecule but it is not detected during sequencing process.</li>
</ol>
<p>Based on the above arguments, I think the probability of not observing a UMI is a mixture of two parts.</p>
<p><span class="math display">\[
Pr(\text{not observe}) = Pr(\text{not observe}|\text{not attach}) Pr(\text{not attach}) + 
Pr(\text{not observe|attach}) Pr(\text{attach}))
\]</span>
Apparently <span class="math inline">\(Pr(\text{not observe}|\text{not attach}) = 1\)</span> and <span class="math inline">\(Pr(\text{not observe|attach})\)</span> should be considered as observing zero count according to a Poisson distribution.</p>
<p>Meanwhile, <span class="math inline">\(Pr(\text{not attach})\)</span> itself is also a mixture of two parts:</p>
<p><span class="math display">\[
Pr(\text{not attach}) = Pr(\text{not attach|gene not express}) Pr(\text{gene not express}) + 
Pr(\text{not attach|gene express}) Pr(\text{gene express})
\]</span>
Of course <span class="math inline">\(Pr(\text{not attach|gene not express}) = 1\)</span></p>
<p>For a gene with <span class="math inline">\(m_i\)</span> copies of RNA molecules.
There are <span class="math inline">\({K \choose m_i}\)</span> ways of combinations of UMI markers and RNA molecules.
Since every combination is equally possible, <span class="math inline">\(Pr(\text{attach}) = \frac{1}{K \choose m_i} = \frac{m_i!(K - m_i)!}{K!}\)</span>.</p>
<p>For example, when <span class="math inline">\(m_i = 1\)</span>, then <span class="math inline">\(Pr(\text{attach}) = \frac{1}{K}\)</span>.
In this case, <span class="math inline">\(1 - \frac{1}{K} = 1 - Pr(\text{attach}) = Pr(\text{not attach})\)</span></p>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<p>I do not think that this conversion makes sense and necessary.
Attaching UMI and RNA molecules is independent only between genes.
Thus, the power of <span class="math inline">\(m_i\)</span> in the equation does not make sense.</p>
</div>
<div id="reference" class="section level1 unnumbered">
<h1>Reference</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Grun2014" class="csl-entry">
Grün, Dominic, Lennart Kester, and Alexander van Oudenaarden. 2014. <span>“<span class="nocase">Validation of noise models for single-cell transcriptomics.</span>”</span> <em>Nat. Methods</em> 11 (6): 637–40. <a href="https://doi.org/10.1038/nmeth.2930">https://doi.org/10.1038/nmeth.2930</a>.
</div>
<div id="ref-Hashimshony2016" class="csl-entry">
Hashimshony, Tamar, Naftalie Senderovich, Gal Avital, Agnes Klochendler, Yaron de Leeuw, Leon Anavy, Dave Gennert, et al. 2016. <span>“<span class="nocase">CEL-Seq2: sensitive highly-multiplexed single-cell RNA-Seq.</span>”</span> <em>Genome Biol.</em> 17 (April): 77. <a href="https://doi.org/10.1186/s13059-016-0938-8">https://doi.org/10.1186/s13059-016-0938-8</a>.
</div>
<div id="ref-Hashimshony2012" class="csl-entry">
Hashimshony, Tamar, Florian Wagner, Noa Sher, and Itai Yanai. 2012. <span>“<span class="nocase">CEL-Seq: single-cell RNA-Seq by multiplexed linear amplification.</span>”</span> <em>Cell Rep.</em> 2 (3): 666–73. <a href="https://doi.org/10.1016/j.celrep.2012.08.003">https://doi.org/10.1016/j.celrep.2012.08.003</a>.
</div>
</div>
</div>
</div>

    
    
    

    
    
        <h4 class="page-header">Comments</h4>
        <div id="disqus_thread"></div>
<script type="application/javascript">
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "username" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
    

</main>

        <footer>
            <p class="copyright text-muted">© All rights reserved. Powered by <a href="https://gohugo.io">Hugo</a> and <a href="https://github.com/calintat/minimal">Minimal</a>.</p>
        </footer>

        

        
    </body>

</html>

